ctimer.h

```diff
@@ -0,0 +1,440 @@
+ * @sa ctimer_start
+ * @sa ctimer_stop
+ */
+static inline
+void ctimer_measure(
+    ctimer_t * t                /**<[in,out] stopwatch pointer */
+) {
+    timespec_sub(&t->elapsed, t->end, t->start);
+}
+
+
+/**
+ * Measure elapsed time of `ctimer_t` stopwatch in s+ns and *add* it to the
+ * `elapsed` timer.
+ *
+ * @warning The `elapsed` field of the input stopwatch must be properly
+ * initialized (e.g. with `ctimer_reset()`) before calling `ctimer_lap()`.
+ *
+ * @warning The stopwatch must be started and stopped before the elapsed time is
+ * measured.
+ *
+ * @note When measuring the execution time of a single, contiguous chunk of a
+ * program, use `ctimer_measure()`.
+ *
+ * @sa ctimer_reset
+ * @sa ctimer_measure
+ * @sa ctimer_start
+ * @sa ctimer_stop
+ */
+static inline
+void ctimer_lap(
+    ctimer_t * t                /**<[in,out] stopwatch pointer */
+) {
+    /* elapsed += end - start */
+    timespec_add(&t->elapsed, t->elapsed, t->end);
+    timespec_sub(&t->elapsed, t->elapsed, t->start);
+}
+
+
+/**
+ * Zero out the `elapsed` timer of a `ctimer_t` stopwatch.
+ *
+ * @sa ctimer_lap
+ */
+static inline
+void ctimer_reset(
+    ctimer_t * t                /**<[in,out] stopwatch pointer */
+) {
+    t->elapsed = (struct timespec){0};
+}
+
+
+/**
+ * Start a `ctimer_t` stopwatch.  Sets the the `start` timer of the stopwatch.
+ *
+ * @sa ctimer_stop
+ * @sa ctimer_measure
+ * @sa ctimer_lap
+ */
+static inline
+void ctimer_start(
+    ctimer_t * t                /**<[in,out] stopwatch pointer */
+) {
+    clock_gettime(CLOCK_MONOTONIC, &t->start);
+}
+
+
+/**
+ * Stop a `ctimer_t` stopwatch.  Sets the `end` timer of the stopwatch.
+ *
+ * @note If the `CTIMER_MEASURE_ON_STOP` preprocessor macro is defined _before_
+ * the `ctimer/ctimer.h` library is included in a program, then `ctimer_stop`
+ * also calculates the elapsed time between `start` and `end` and stores it in
+ * the `elapsed` field.
+ *
+ * @sa ctimer_start
+ * @sa ctimer_measure
+ * @sa ctimer_lap
+ */
+static inline
+void ctimer_stop(
+    ctimer_t * t                /**<[in,out] stopwatch pointer */
+) {
+    clock_gettime(CLOCK_MONOTONIC, &t->end);
+#ifdef CTIMER_MEASURE_ON_STOP
+    ctimer_measure(t);
+#endif
+}
+
+
+/**
+ * Print a line with the `elapsed` time of a `ctimer_t` stopwatch in seconds.
+ *
+ * The line is printed as:
+ * ```
+ * Time(<label>) = XX.XXXXXXXXX sec
+ * ```
+ *
+ * @note This utility function is meant to facilitate standardized printing of
+ * elapsed times which may then be parsed by external programs.
+ *
+ * @sa ctimer_measure
+ * @sa ctimer_lap
+ */
+static inline
+void ctimer_print(
+    ctimer_t const   t,         /**<[in] stopwatch pointer */
+    char     const * label      /**<[in] label/description for printed time */
+) {
+    if ((label != NULL) && (label[0] != '\0'))
+        printf("Time(%s) = ", label);
+    else
+        printf("Time = ");
+
+    printf("%ld.%09ld sec\n", (long)t.elapsed.tv_sec, t.elapsed.tv_nsec);
```

Comment:
As a minor point, on one system where I'm testing this PR, `CLOCK_MONOTONIC` has only microsecond resolution, so all timings have three 0's at the end.  Timer resolution can vary between systems, and although you could use `clock_getres` to get the clock resolution and format the output, it might not be worth the effort.

Reply:
Thanks @neboat. 

@ailiop I think these requested changes await your approval? Or should I approve them? I'm not sure if I have clicked "resolve conversation" before...

Reply:
@neboat Thank you, I merged your fix and added a note in the `ctimer.h` documentation regarding the resolution of the printed time.

I'm sorry this fell off my radar.

Reply:
@behoppe Thanks for following up.  I'd lost track of this PR.